File: web/src/api/lg/pay.js
import service from '@/utils/request'

export const createPay = (data) => {
  return service({
    url: '/pay/createPay',
    method: 'post',
    data
  })
}

export const deletePay = (data) => {
  return service({
    url: '/pay/deletePay',
    method: 'delete',
    data
  })
}

export const deletePayByIds = (data) => {
  return service({
    url: '/pay/deletePayByIds',
    method: 'delete',
    data
  })
}

export const updatePay = (data) => {
  return service({
    url: '/pay/updatePay',
    method: 'put',
    data
  })
}

export const findPay = (params) => {
  return service({
    url: '/pay/findPay',
    method: 'get',
    params
  })
}

export const getPayList = (params) => {
  return service({
    url: '/pay/getPayList',
    method: 'get',
    params
  })
}

File: server/router/lg/pay.go
package lg

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type PayRouter struct {
}

func (s *PayRouter) InitPayRouter(Router *gin.RouterGroup) {
	payRouter := Router.Group("pay").Use(middleware.OperationRecord())
	payRouterWithoutRecord := Router.Group("pay")
	var payApi = v1.ApiGroupApp.LgApiGroup.PayApi
	{
		payRouter.POST("createPay", payApi.CreatePay)             // 新建Pay
		payRouter.DELETE("deletePay", payApi.DeletePay)           // 删除Pay
		payRouter.DELETE("deletePayByIds", payApi.DeletePayByIds) // 批量删除Pay
		payRouter.PUT("updatePay", payApi.UpdatePay)              // 更新Pay
	}
	{
		payRouterWithoutRecord.GET("findPay", payApi.FindPay)       // 根据ID获取Pay
		payRouterWithoutRecord.GET("getPayList", payApi.GetPayList) // 获取Pay列表
	}
}
